ci/ray_ci/test_utils.py

```diff
@@ -34,5 +41,48 @@ def _mock_subprocess_run(
         docker_login("docker_ecr")
 
 
+def _make_test(name: str, state: str, team: str) -> Test:
+    return Test(
+        {
+            "name": name,
+            "state": state,
+            "team": team,
+        }
+    )
+
+
+@mock.patch("ray_release.test.Test.gen_from_s3")
+def test_get_flaky_test_names(mock_gen_from_s3):
+    mock_gen_from_s3.side_effect = (
+        [
+            _make_test("darwin://test_1", "flaky", "core"),
+            _make_test("darwin://test_2", "flaky", "ci"),
+        ],
+    )
+    flaky_test_names = get_flaky_test_names(
+        prefix="darwin:",
+    )
+    assert flaky_test_names == ["//test_1", "//test_2"]
+
+
+@mock.patch("ray_release.test.Test.gen_from_s3")
+def test_filter_out_flaky_tests(mock_gen_from_s3):
+    mock_gen_from_s3.side_effect = (
+        [
+            _make_test("darwin://test_1", "flaky", "core"),
+            _make_test("darwin://test_2", "flaky", "ci"),
+            _make_test("darwin://test_3", "passing", "core"),
+            _make_test("darwin://test_4", "passing", "ci"),
+        ],
+    )
+
+    test_targets = ["//test_1", "//test_2", "//test_3", "//test_4"]
+    sys.stdout = io.StringIO()
+    filter_out_flaky_tests(io.StringIO("\n".join(test_targets)), sys.stdout, "darwin:")
+
+    filtered_test_targets = sys.stdout.getvalue().strip()
```

Comment:
why `strip()`?

Reply:
oh I was just getting rid of newline char to compare it with expected result, but I changed the expected value to include new char as well so I'm getting rid of `.strip()` now.